gems/pending/disk/modules/AzureBlobDisk.rb

```diff
@@ -72,7 +72,7 @@ def blob_read(start_byte, length)
     ret = @storage_acct.get_blob_raw(@container, @blob, key, options)
 
     content_md5  = ret.headers[:content_md5].unpack("m0").first.unpack("H*").first
-    returned_md5 = Digest::MD5.hexdigest(ret.body)
+    returned_md5 = Rails.application.config.digest_class.hexdigest(ret.body)
```

Comment:
It seems the only way this can work for all configurations is to remove the MD5 check.
The code: `Rails.application.config.digest_class.hexdigest` can't be used in this context for a number of reasons:

- This code can run outside of a Rails context, so `Rails.application` may not be available.

- Whenever `Rails.application.config.digest_class.hexdigest` is configured to use something other than MD5, the check will fail, effectively preventing Azure SSA in those environments.